### DISCUSSION TITLE
tailwind.config.js not work at monorepo

### DISCUSSION THREAD

**Discussion by Yueyanc** [tailwindlabs/tailwindcss#14630] [Help]
**What version of Tailwind CSS are you using?**

v3.4.13

**What build tool (or framework if it abstracts the build tool) are you using?**

react@18.3.1

**What version of Node.js are you using?**

v20.11.1

**What browser are you using?**

electron

**What operating system are you using?**

Windows

**Reproduction URL**

https://github.com/Yueyanc/template/tree/electron-template

**Describe your issue**

If I set tailwind.config.js in the root directory of the monorepo project, the configuration takes effect. However, it does not take effect when I set tailwind.config.js in the root directory of the web project

![image](https://github.com/user-attachments/assets/106ffa5b-3614-47b8-978a-3d5b376c5649)
![image](https://github.com/user-attachments/assets/e2e95372-994b-42cb-95b5-dd6988e4ebe9)
![image](https://github.com/user-attachments/assets/9ea1b518-b6d5-45f2-b69f-c175f4638bd3)

**Comment by RobinMalfait** [comment]
Hey!

A few things I can think of that could cause issues here:

How do you start the application? Do you want to develop the `web` and `electron` projects together or separate?

If you run `npm run dev` in the root, then it looks like it is loading the `vite` configuration from the `Electron` folder, not the web folder.

If you go into the `web` folder directly, and run `npm run dev`, then the page loads for me (without styles) but not a single style is applied. I also noticed that no stylesheet was linked. If I then apply this diff:
```diff
diff --git a/apps/web/index.html b/apps/web/index.html
index 44e484f..156e0ff 100644
--- a/apps/web/index.html
+++ b/apps/web/index.html
@@ -3,6 +3,7 @@
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <link rel="stylesheet" href="./src/index.css" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>Vite + React + TS</title>
   </head>
```
Then I do get the styles without any more changes.


So, the question is, how do you want to develop? If you want to be able to run `npm run dev` from the root. Then you have to make sure that in the `scripts/watch.ts` file you also somehow use the `vite.config.ts` from the `web/` folder.

**Reply by Yueyanc** [reply]
I want tailwind.config.js in the web directory to work when I run npm run dev directly from root, but it's not working at the moment. In addition, if you use npm run dev directly under the web directory, the web project uses the api exposed in electron preload.js, so the error will cause the view cannot be displayed, but tailwind.config.js is valid
I took a look at the source code for tailwindcss and found that the logic for finding tailwind.config.js is direct path.resolve
![image](https://github.com/user-attachments/assets/95c43940-3a43-4773-907f-4a8255243647)
![image](https://github.com/user-attachments/assets/3299cc43-38e0-46f8-961d-5a5670f86bb0)

**Comment by Yueyanc** [comment]
I resolve the issue by

![image](https://github.com/user-attachments/assets/931fce06-3525-4f2f-b45c-83aced7dd947)

![image](https://github.com/user-attachments/assets/a8bda8ae-f778-4dd0-be8b-be5d0e732651)

But I don't think that's the best solution
